### PR TITLE
Add React guideline

### PR DESCRIPTION
**Pull request by FourthOf5**

### CODE REVIEW

File: react/styleguide.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+## Syntax
+
+1. #### Naming Style **[⬆](#table-of-content)**
+
+   * Always use `.jsx` | `.tsx` extensions.
+   * Use UpperCamelCase as a components name.
+   * Use lowerCamelCase for prop names.
+   * It's recommended to use suffixes `Component` or `Container` for explicit definition of "dumb" and "smart" components: `UserProfileComponent` and `UserProfileContainer`.
+   * Component name should be the same as the file name. This means that the use of `index` files is forbidden for components. It's easier to find components by filename rather than by folder name.
+   * Component filenames should be in lowerCamelCase. It's allowed to use dots (`.`) to explicitly mention type of component: `dashboard.component.tsx`, `userProfile.container.tsx`.
```
Comment (artyorsh) [inline react/styleguide.md:32]: A common practice is to use UpperCamelCase, but correct me if I'm wrong. Personally, I prefer dash-case over both mentioned options

Comment (FourthOf5) [reply react/styleguide.md:32]: Agreed with the UpperCamelCase. lowerCamelCase was left from the react-native style guide. 
Dash-case looks a bit irregular to me, so I would stick to the UpperCamelCase

Comment (malashkevich) [reply react/styleguide.md]: lowerCamelCase was for the TS at all. Please take a look

File: react/styleguide.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+   ```
+   * Before self-closing `/>` tag should be a single space.
+
+1. #### Line breaks **[⬆](#table-of-content)**
+
+   * Component with a single prop without children should take a single line.
+   * Component with more than one prop or with children must be multiline
+     * Each prop should start from the new line
+     * Children should start from the new line
+     * Closing bracket of the opening tag `>` and self-closing `/>` tag should stick to the line with last prop.
```
[the remark this answers is not in the document]
Comment (artyorsh) [reply react/styleguide.md:55]: @FourthOf5 generally you should never have **closing** tags if the component has no children. In this case, it's visually better to have the tag on a same line as its last property.
```jsx
<UserProfile 
  name='Boris'
  lastName='Blade'>
  <NestedComponent />
  <NestedComponent />
</UserProfile>    
```

On the other hand, **self-closing** tags can be used only when there is no children. In this case, I would suggest using the syntax from the previous comment

[the remark this answers is not in the document]
Comment (artyorsh) [reply react/styleguide.md:55]: @malashkevich did I? 😄 I could miss that in PR but I believe I had complained about this internally multiple times before :)

File: react/styleguide.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+         onPress={onPress}>
+         {buttonText}
+       </Button>
+       ```
+     * Self-closing `/>` tag should be placed on the new line
+       ```tsx
+       <UserProfile
+         name='Boris'
+         lastName='Blade' 
+       />
```
Comment (malashkevich) [inline react/styleguide.md:68]: Does it mean we have to apply this to react-native? I would say yes

Comment (malashkevich) [reply react/styleguide.md:68]: But I'm still thinking this is not a good idea to have different styling depending on the component structure.

**Review by malashkevich — APPROVED** [review]
ok to me, but with comments

File: react/styleguide.md (hunk 3 above)
Comment (FourthOf5) [reply react/styleguide.md:68]: @malashkevich  I agree. Then we need to decide for both cases, whether we place the clothing bracket on the same line or on the new line

**Pull request merged**